app/controllers/comments_controller.rb

```diff
@@ -0,0 +1,14 @@
+class CommentsController < ApplicationController
+  def create
+    puts params
+    @comment = current_user.comments.new(params.require(:comment).permit(:text))
+    @comment.post = Post.find(params[:post_id])
+    if @comment.save
+      flash[:success] = 'Comment created successfully'
+    else
+      flash[:error] = 'Something went wrong'
+      puts 'error'
+    end
+    redirect_to user_post_path(@comment.post.author, @comment.post)
+  end
```

Comment:
- [ ] [OPTIONAL] Inside your post controller, I would like to recommend you to implement `params`  method you can find more details about it right  [here](`https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/lessons/Controllers_lesson.md#:~:text=Pay%20special%20attention%20to%20the%20strong%20parameters%20description%20in%20Rails%20Guides`) :+1: 

```ruby
  private 
  def comment_params
    params.require(:comment).permit(:text)
  end
```